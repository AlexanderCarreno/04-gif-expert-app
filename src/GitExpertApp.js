//rafc
import React, { useState } from 'react';

const GitExpertApp = ({}) => {
    //const  categories = ['One Punch', 'Guerra de magos', 'Demon Slayer'];

    const [categories, setCategories] = useState(['One Punch', 'Guerra de magos', 'Demon Slayer']);

    const handleAdd = () => {
        //const newElement = ['My Hero Academia'];
        //setCategories([...categories, ...newElement]);

        //setCategories([...categories, 'My Hero Academia']);

        setCategories( cats => [...cats, 'My Hero Academia'])
    }


    return(
        <>
            <h2>GitExpertApp</h2>
            <hr/>

            <button onClick={ handleAdd }>Agregar</button>

            <ol>
                { 
                    categories.map( category =>{
                        return <li key={ category } >{category}</li>
                    })
                }
            </ol>
        </>
    )
}

export default GitExpertApp;