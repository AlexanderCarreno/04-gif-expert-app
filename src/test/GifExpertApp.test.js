import React from 'react';
import { shallow } from 'enzyme';
import GitExpertApp from '../GifExpertApp';

describe('debe mostrar el componente <GitExpertApp /> correctamente', () => { 
    
    let wrapper = shallow( <GitExpertApp />);

    beforeEach( ()=>{
        wrapper = shallow( <GitExpertApp />);
    });

    test('debe mostrar <GitExpertApp /> corractamente', () => { 
        expect( wrapper ).toMatchSnapshot();
     });


     test('debe de mostrar una lista de categorias', () => { 
         const categories = ['One Punch', 'Dragon Ball'];
         const wrapper = shallow(<GitExpertApp defaultCategories={ categories } />);


         expect( wrapper ).toMatchSnapshot();
         expect( wrapper.find('GifGrid').length  ).toBe( categories.length );
      });
});