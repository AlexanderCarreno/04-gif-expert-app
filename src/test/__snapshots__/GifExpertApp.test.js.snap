// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe mostrar el componente <GitExpertApp /> correctamente debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`debe mostrar el componente <GitExpertApp /> correctamente debe mostrar <GitExpertApp /> corractamente 1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
